mirascope/core/openai/_utils/_setup_call.py

```diff
@@ -81,4 +127,4 @@ def setup_call(
         client = AsyncOpenAI() if inspect.iscoroutinefunction(fn) else OpenAI()
     create = client.chat.completions.create
 
-    return create, prompt_template, messages, tool_types, call_kwargs
+    return create, prompt_template, messages, tool_types, call_kwargs  # pyright: ignore [reportReturnType]
```

Comment:
same thing here, why can't we remove the type ignore?

Reply:
There are two issues.

The stream argument type is `bool` plus `NotGiven`, which results in a type error.
Also, the return value is expected to be a `Generator`, but it returns an Iterator-like type. For this reason, it is necessary to ignore the type.
Since `Generator` is a subclass of Iterator, strictly speaking, it is necessary to add or wrap the missing methods as a generator.

Reply:
Fixed